### PR TITLE
Add CreateThread to avoid manual try-catch

### PR DESCRIPTION
**Pull request by PragmaTwice**
Add `CreateThread`, `ThreadJoin` and `ThreadDetach` to wrap `std::thread` operations, to avoid manual try-catch.

### CODE REVIEW

File: src/commands/cmd_replication.cc [hunk 1]
```diff
@@ -271,48 +273,52 @@ class CommandFetchFile : public Commander {
...
-                    << " ms since of sending files too quickly";
-          usleep(shortest - duration);
-        }
-      }
-      auto now = static_cast<time_t>(Util::GetTimeStamp());
-      svr->storage_->SetCheckpointAccessTime(now);
-      svr->DecrFetchFileThread();
-    });
-    t.detach();
+    std::thread t =
```
Comment (torwig) [inline src/commands/cmd_replication.cc]: I think you can use `auto` instead of `std::thread` here too.

**Review by torwig — DISMISSED** [review]
LGTM

**Review by git-hulk — DISMISSED** [review]

File: src/commands/cmd_replication.cc (hunk 1 above)
Comment (PragmaTwice) [reply src/commands/cmd_replication.cc]: done

**Review by torwig — APPROVED** [review]

**Review by caipengbo — APPROVED** [review]

**Comment by PragmaTwice** [comment]
Thanks all. Merging...

**Pull request merged**